server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/api/MailQueueView.java

```diff
@@ -34,6 +34,10 @@
 
     CompletableFuture<Void> deleteMail(Mail mail);
 
+    CompletableFuture<Long> delete(DeleteCondition deleteCondition);
+
+    CompletableFuture<Boolean> isDeleted(Mail mail);
```

Comment:
I don't like negatively-named method, it leads to double (or more) negation.
What about isPresent instead ?

Reply:
It really depend on the usage, but there are none yet (in this PR).

It can be modified later.

Reply:
it's not.
with `isDeleted` you will write `if(isDeleted())` and `if(!isDeleted())` (double negation)
with `isPreset` you will write `if(!isPresent())` and `if(isPresent)` (no double negation)